### PR TITLE
Add json output to install and related commands

### PR DESCRIPTION
**Pull request by adleong**
We add support for the `--output/-o` flag in linkerd install and related commands. The supported output formats are yaml (default) and json. Kubectl is able to accept both of these formats which means that the output can be piped into kubectl regardless of which output format is used.

The full list of install related commands which we add json support to is:

* linkerd install
* linkerd prune
* linkerd upgrade
* linkerd uninstall
* linkerd viz install
* linkerd viz prune
* linkerd viz uninstall
* linkerd multicluster install
* linkerd multicluster prune
* linkerd multicluster uninstall
* linkerd jaeger install
* linkerd jaeger prune
* linkerd jaeger uninstall

### CODE REVIEW

File: cli/cmd/install.go [hunk 1]
```diff
@@ -358,11 +363,34 @@ func renderCRDs(w io.Writer, options valuespkg.Options) error {
...
+			_, err = w.Write(append(bytes, '\n'))
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	} else if format == "yaml" {
+		_, err = w.Write(buf.Bytes())
+		return err
+	}
```
Comment (alpeb) [inline cli/cmd/install.go:389]: Nit:
```suggestion
	if format == "yaml" {
		_, err = w.Write(buf.Bytes())
		return err
	}
```

File: cli/cmd/install.go [hunk 2]
```diff
@@ -389,8 +417,31 @@ func renderControlPlane(w io.Writer, values *l5dcharts.Values, valuesOverrides m
...
```
Comment (alpeb) [inline cli/cmd/install.go:443]: Same nit:
```suggestion
	if format == "yaml" {
		_, err = w.Write(buf.Bytes())
		return err
	}
```

File: jaeger/cmd/install.go [hunk 3]
```diff
@@ -190,6 +196,29 @@ func render(w io.Writer, valuesOverrides map[string]interface{}, registry string
...
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	} else if format == "yaml" {
+		_, err = w.Write(buf.Bytes())
+		return err
+	}
+	return fmt.Errorf("unsupported format %s", format)
```
Comment (alpeb) [inline jaeger/cmd/install.go:223]: This is ripe for moving into a library given it's repeated almost verbatim so many times.

File: cli/cmd/install.go [hunk 4]
```diff
@@ -358,11 +363,34 @@ func renderCRDs(w io.Writer, options valuespkg.Options) error {
 		return err
 	}
 
-	_, err = w.Write(buf.Bytes())
-	return err
+	if format == "json" {
```
Comment (mateiidavid) [inline cli/cmd/install.go:366]: In `cli/cmd/root.go` we have a `jsonOutput` constant. Should we use that here?

https://github.com/linkerd/linkerd2/blob/b028db78a2f0d33b1424adf62bc50cab2e48916e/cli/cmd/root.go#L26-L28

should we also add one for `yaml`?

File: viz/cmd/uninstall.go [hunk 5]
```diff
@@ -86,18 +89,26 @@ func uninstallRunE(ctx context.Context) error {
...
 	r := resource.NewNamespaced(ty.APIVersion, ty.Kind, meta.Name, meta.Namespace)
-	if err := r.RenderResource(os.Stdout); err != nil {
-		return fmt.Errorf("error rendering Kubernetes resource: %w", err)
+	if format == "json" {
+		if err := r.RenderResourceJSON(os.Stdout); err != nil {
+			return fmt.Errorf("error rendering Kubernetes resource: %w", err)
+		}
+		return nil
+	} else if format == "yaml" {
+		if err := r.RenderResourceJSON(os.Stdout); err != nil {
```
Comment (mateiidavid) [inline viz/cmd/uninstall.go:108]: Should probably call `RenderResource` here, right? Seems like both of them call `RenderResourceJSON`

File: cli/cmd/install.go [hunk 6]
```diff
@@ -148,7 +152,7 @@ A full list of configurable values can be found at https://artifacthub.io/packag
 			if crds {
 				// The CRD chart is not configurable.
```
Comment (the-wondersmith) [inline cli/cmd/install.go:153]: Not strictly about the changes at hand, but in keeping with the [Boy Scout Rule](https://97-things-every-x-should-know.gitbooks.io/97-things-every-programmer-should-know/content/en/thing_08/), this isn't (strictly speaking) accurate any more [ref](https://github.com/linkerd/linkerd2/blob/main/charts/linkerd-crds/values.yaml#L1).

Should we remove the comment and/or todo?

File: viz/cmd/uninstall.go (hunk 5 above)
Comment (adleong) [reply viz/cmd/uninstall.go]: good catch, thanks!

File: cli/cmd/install.go (hunk 6 above)
Comment (adleong) [reply cli/cmd/install.go:151]: Good point, but let's keep unrelated improvements in their own PRs.

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**